backend/tournesol/views/video.py

```diff
@@ -139,32 +141,16 @@ def get_serializer_class(self):
             return VideoSerializerWithCriteria
         return VideoSerializer
 
-    def create(self, request, *args, **kwargs):
-        """
-        Add a video to the db if it does not already exist
-        """
-        video_id = request.data.get("video_id")
-        if not video_id:
-            return Response('No video_id given', status=status.HTTP_400_BAD_REQUEST)
-        if len(video_id) != 11:
-            return Response("video_id is not valid", status=status.HTTP_400_BAD_REQUEST)
-        if Video.objects.filter(video_id=video_id):
-            Response(
-                "The video with the id : {id} is already registered".format(
-                    id=request.data["video_id"]),
-                status=status.HTTP_400_BAD_REQUEST
-            )
```

Comment:
A `return` statement was missing here to avoid a call to the Youtube API when the video already exists in our database. 

By overriding `perform_create()` instead of `create()`, the `VideoSerializer` is initialized automatically, and we can avoid redundant validation here.